src/workers/postUpdated.ts

```diff
@@ -567,10 +567,15 @@ const getSourcePrivacy = async ({
       query = query.where('source.id = :id', { id: data?.source_id });
     }
 
-    const source = await query.getOne();
-    return source?.private;
+    const source = await query.getOneOrFail();
+
+    return source.private;
   } catch (err) {
-    logger.error({ data, err }, 'failed find source for post');
+    const sourcePostError = new Error('failed to find source for post');
+
+    logger.error({ data, err }, sourcePostError.message);
+
+    throw sourcePostError;
```

Comment:
We now throw if source is not found.

Only place where `source_id` is not sent from yggdrasil is on freeform (due to existing on API) but we have that case covered.

Will monitor of course after deploy for this errors.